apps/asap-server/src/handlers/users/update.ts

```diff
@@ -20,16 +20,16 @@ export const handler: Handler = lambda.http(
       id: string;
     };
 
+    if (user.id !== params.id) {
```

Comment:
I don't really agree with changing the ordering here. We need to validate that the body schema is valid before doing the more advanced authorisations, like the one on #465